oauth/provider/ldap/ldap.go

```diff
@@ -0,0 +1,90 @@
+package ldap
+
+import (
+	"fmt"
+	. "github.com/ory-am/hydra/oauth/provider"
+	"gopkg.in/ldap.v2"
+	"crypto/tls"
+)
+
+type ldapconf struct {
+	id       string
+	host     string
+	port     string
+	tls      bool
+	base_dn  string
+	bind_dn  string
+	uid      string
+	password string
+	filter   string
+}
+
+func New(id, client, secret, redirectURL string) *ldapconf {
+	return &ldapconf{
+		id:  id,
+		host: "ldap.forumsys.com",
+		tls: false,
+		port: "389",
+		bind_dn: "cn=read-only-admin,dc=example,dc=com",
+		base_dn: "dc=example,dc=com",
+		uid: "uid",
+		password: "password",
+		filter: "(objectClass=organizationalPerson)",
+	}
```

Comment:
Actually, the whole thing should be configurable. Maybe don't define `New` at all and simply make `ldapconf` public and rename it to e.g. `LDAPProvider`.